File: soccer/gameplay/plays/Stopped.cpp
#include "Stopped.hpp"

using namespace std;

REGISTER_PLAY_CATEGORY(Gameplay::Plays::Stopped, "Restarts")

Gameplay::Plays::Stopped::Stopped(GameplayModule *gameplay):
	Play(gameplay),
	_idle(gameplay),
	_left(gameplay, Behaviors::Fullback::Left),
	_right(gameplay, Behaviors::Fullback::Right)
{
	set<Robot *> available = _gameplay->robots();
	
	_left.otherFullbacks.insert(&_right);
	_right.otherFullbacks.insert(&_left);
	
	_robots = available;
	
	_left.assign(available);
	_right.assign(available);
	_idle.assign(available);
}

float Gameplay::Plays::Stopped::score ( Gameplay::GameplayModule* gameplay )
{
	return gameplay->state()->gameState.stopped() ? 0 : INFINITY;
}

bool Gameplay::Plays::Stopped::run()
{
	_idle.run();
	_left.run();
	_right.run();
	
	return true;
}

File: soccer/gameplay2/plays/stopped.py
import play
import behavior
import tactics


# when we get the Stopped command from the referee,
# we run this play.  See the rules to see what we're allowed to do while the game is stopped
class Stopped(play.Play):
    def __init__(self):
        super().__init__(continuous=True)
        self.add_transition(behavior.Behavior.State.start,
            behavior.Behavior.State.running,
            lambda: True,
            'immediately')


    def score(self):
        return 0 if main.game_state().is_stopped() else float("inf")


    def on_enter_running(self):
        left = tactics.positions.fullback.Fullback()
        self.add_subbehavior(left, 'left_fullback', required=False, priority=50)

        right = tactics.positions.fullback.Fullback()
        self.add_subbehavior(right, 'right_fullback', required=False, priority=49)

        idle = tactics.circle_near_ball.CircleNearBall()
        self.add_subbehavior('circle_up', idle, required=False, priority=1)


    def on_exit_running(self):
        for name in ['circle_up', 'right_fullback', 'left_fullback']:
            self.remove_subbehavior(name)
